pyo3-build-config/src/impl_.rs

```diff
@@ -638,6 +627,20 @@ pub fn cross_compiling(
         // No cross-compiling environment variables set; try to determine if this is a known case
         // which is not cross-compilation.
 
+        if host.starts_with(&target_triple) {
```

Comment:
So this is basically subsuming the `host == target` condition with the idea that setting the environment variables should force cross compilation even if host and target are exactly equal?